astrocut/tests/test_footprint_cutouts.py

```diff
@@ -0,0 +1,154 @@
+import re
+import pytest
+
+from astrocut.exceptions import InvalidQueryError
+from astrocut.footprint_cutouts import (cube_cut_from_footprint, _extract_sequence_information, _s_region_to_polygon, 
+                                        get_caom_ffis, _ffi_intersect, ra_dec_crossmatch, _create_sequence_list)
+from astropy.coordinates import SkyCoord
+from astropy.io import fits
+from astropy.table import Table
+from spherical_geometry.polygon import SphericalPolygon
+
+
+def check_output_file(path, ffi_type, sequences=[]):
+    """Helper function to check the validity of output cutout files"""
+    # Check that cutout path point to valid FITS file
+    tpf = fits.open(path)
+    tpf_table = tpf[1].data
+
+    # SPOC cutouts have 1 extra columns in EXT 1
+    ncols = 12 if ffi_type == 'SPOC' else 11
+    assert len(tpf_table.columns) == ncols
+    assert tpf_table[0]['FLUX'].shape == (5, 5)
+
+    # Check that sector matches a provided sequence
+    if sequences:
+        assert tpf[0].header['SECTOR'] in sequences
+
+    tpf.close()
+
+
+def test_s_region_to_polygon_unsupported_region():
+    """Test that ValueError is raised if s_region is not a polygon"""
+    s_region = 'CIRCLE'
+    err = 'Unsupported S_Region type.'
+    with pytest.raises(ValueError, match=err):
+        _s_region_to_polygon(s_region)
+
+
+@pytest.mark.parametrize("lon, lat, center, expected", [
+    ((345, 355, 355, 345), (-15, -15, -5, -5), (350, -10), True),  # intersecting
+    ((335, 345, 345, 335), (-15, -15, -5, -5), (340, -10), False),  # non-intersecting
+    ((340, 350, 350, 340), (-15, -15, -5, -5), (345, -10), True),  # edge object that intersects
+    ((340, 349, 349, 340), (-15, -15, -5, -5), (345, -10), False),  # edge object that does not intersect
+])
+def test_ffi_intersect(lon, lat, center, expected):
+    """Test that FFI intersection with cutout outputs proper results."""
+    # SphericalPolygon object for cutout
+    cutout_sp = SphericalPolygon.from_radec(lon=(350, 10, 10, 350),
+                                            lat=(-10, -10, 10, 10),
+                                            center=(0, 0))
+
+    # Create a SphericalPolygon with the parametrized lon, lat, and center
+    polygon = SphericalPolygon.from_radec(lon=lon, lat=lat, center=center)
+
+    # Create a table with this polygon
+    polygon_table = Table(names=['polygon'], dtype=[SphericalPolygon])
+    polygon_table['polygon'] = [polygon]
+
+    # Perform the intersection check
+    intersection = _ffi_intersect(polygon_table, cutout_sp)
+
+    # Assert the intersection result matches the expected value
+    assert intersection.value[0] == expected
```

Comment:
maybe this assertion should be done with `is` instead?

```suggestion
    assert intersection.value[0] is expected
```

Reply:
Using `is` fails because `intersection.value[0]` is a `numpy.bool_` object.

Reply:
oh, gotcha. okay!